descartes_planner/src/planning_graph.cpp

```diff
@@ -630,45 +640,48 @@ bool PlanningGraph::getShortestPath(double &cost, std::list<JointTrajectoryPt>&
     vertex_index_map[i++] = dg_[jv].id;
   }
 
+  // Default the case to a very large number incase there is no valid path
+  // through the graph
   cost = std::numeric_limits<double>::max();
 
-  for (auto start = start_points.begin(); start != start_points.end(); ++start)
-  {
-    // initialize vectors to be used by dijkstra
-    std::vector<JointGraph::vertex_descriptor> predecessors(num_vert);
-    std::vector<double> weights(num_vert, std::numeric_limits<double>::max());
+  // initialize vectors to be used by dijkstra
+  std::vector<JointGraph::vertex_descriptor> predecessors(num_vert);
+  std::vector<double> weights(num_vert, std::numeric_limits<double>::max());
 
-    dijkstra_shortest_paths(
-        dg_,
-        *start,
-        weight_map(get(&JointEdge::transition_cost, dg_)).distance_map(
-          boost::make_iterator_property_map(weights.begin(), get(boost::vertex_index, dg_))).predecessor_map(
-          &predecessors[0]));
+  dijkstra_shortest_paths(
+      dg_,
+      virtual_vertex, // start from our fake point
+      weight_map(get(&JointEdge::transition_cost, dg_)).distance_map(
+        boost::make_iterator_property_map(weights.begin(), get(boost::vertex_index, dg_))).predecessor_map(
+        &predecessors[0]));
 
-    for (auto end = end_points.begin(); end != end_points.end(); ++end)
+  // Search the ending points for a minimum point
+  for (auto end = end_points.begin(); end != end_points.end(); ++end)
+  {
+    double weight = weights[*end];
+    // if the weight of this path of less than a previous one, replace the return path
+    if (weight < cost)
     {
-      // actual weight(cost) from start_id to end_id
-      double weight = weights[*end];
-
-      // if the weight of this path of less than a previous one, replace the return path
-      if (weight < cost)
+      cost = weight;
+      path.clear();
+      // Add the destination point.
+      auto current = *end;
+      // path.push_front(joint_solutions_map_[vertex_index_map[current]]);
+      // Starting from the destination point step through the predecessor map
+      // until the source point is reached.
+      while (current != virtual_vertex)
```

Comment:
This PR only improves the search component. It does not affect how long it takes to build the graph. The times I mentioned above are the sum of the build and search. The other pull request i added helps with this. Both changes combined brings the solution time on the last graph to 3.5 seconds.